src/main.rs

```diff
@@ -163,35 +138,69 @@ impl BFInterp {
             MetadataOperand => todo!(),
         };
 
-        let mut string_temp;
-        for arg in call.arguments[1..].iter() {
-            match &arg.0 {
-                LocalOperand { name, .. } => {
-                    string_temp =
-                        string.replacen("{}", self.mem.get(&name).unwrap().to_string().as_str(), 1);
-                    string = &string_temp;
-                }
-                ConstantOperand(con_op) => match con_op.as_ref() {
-                    Constant::Int { value, .. } => {
-                        string_temp = string.replacen("{}", value.to_string().as_str(), 1);
-                        string = &string_temp;
+        let mut string = String::new();
+        let mut arg_it = call.arguments[1..].into_iter();
+        match constant {
+            Constant::Array { elements, .. } => {
+                let mut i = 0;
+                while i < elements.len() {
+                    match elements[i].as_ref() {
+                        Constant::Int { value, .. } => {
+                            let char = *value as u8 as char;
+                            if char == '%' {
+                                string.push_str(
+                                    self.arg_to_string(&arg_it.next().unwrap().0).as_str(),
+                                );
+                                i += 2;
+                                continue;
+                            } else if value == &0u64 {
```

Comment:
Can `value` only ever be `%` and `0`? If not, what happens when it has another value.

Reply:
If it's neither then the char get's pushed to the string. Look just below.

Reply:
I was wondering because I didn't see the `i += 1` before. Got it now.